src/platform/datapath_raw.c

```diff
@@ -130,6 +135,10 @@ CxPlatDataPathInitialize(
     CxPlatZeroMemory(*NewDataPath, DatapathSize);
 
     if (UdpCallbacks) {
+        if (UdpCallbacks->Receive == NULL || UdpCallbacks->Unreachable == NULL) {
```

Comment:
This validation should be before allocating the memory above, otherwise you're leaking it on failure it seems.